HousingSearchApi/V1/Infrastructure/Factories/AssetQueryGenerator.cs

```diff
@@ -19,22 +19,67 @@ public AssetQueryGenerator(IQueryBuilder<QueryableAsset> queryBuilder)
 
         public QueryContainer Create<TRequest>(TRequest request, QueryContainerDescriptor<QueryableAsset> q)
         {
+            //Default to search endpoint
             GetAssetListRequest assetListRequest = request as GetAssetListRequest;
+
             if (assetListRequest == null)
                 throw new ArgumentNullException($"{nameof(request).ToString()} shouldn't be null.");
 
-            return _queryBuilder
-                .WithWildstarQuery(assetListRequest.SearchText,
-                    new List<string> { "assetAddress.addressLine1", "assetAddress.postCode", "assetAddress.uprn" })
-                .WithExactQuery(assetListRequest.SearchText,
-                    new List<string>
-                    {
-                        "assetAddress.addressLine1",
-                        "assetAddress.uprn",
-                        "assetAddress.postCode"
-                    })
-                .WithFilterQuery(assetListRequest.AssetTypes, new List<string> { "assetType" })
-                .Build(q);
+            if (request.GetType() == typeof(GetAssetListRequest))
+            {
+                //This is so assets search endpoint works as before
+                return _queryBuilder
+                    .WithWildstarQuery(assetListRequest.SearchText,
+                        new List<string> { "assetAddress.addressLine1", "assetAddress.postCode", "assetAddress.uprn" })
+                    .WithExactQuery(assetListRequest.SearchText,
+                        new List<string>
+                        {
+                            "assetAddress.addressLine1",
+                            "assetAddress.uprn",
+                            "assetAddress.postCode"
+                        })
+                    .WithFilterQuery(assetListRequest.AssetTypes, new List<string> { "assetType" })
+                    .Build(q);
+            }
+            else
+            {
+                if (assetListRequest.SearchText != null && assetListRequest.SearchText.Length > 0)
```

Comment:
Code is being repeated in this way - could you please structure the conditions in a different way so we don't repeat the return statement twice.